tts/src/androidMain/kotlin/nl/marc_apps/tts/TextToSpeechAndroid.kt

```diff
@@ -192,6 +192,9 @@ internal class TextToSpeechAndroid(private var tts: AndroidTTS?) : TextToSpeechI
                     cont.resumeWithException(error)
                 }
             }
+            cont.invokeOnCancellation {
+                stop()
```

Comment:
The `stop()` method removes everything that's in the queue. This might cause unintended behaviour:
```Kotlin
launch(firstPersonJob) {
    textToSpeech.say("Hello!")
}

launch(secondPersonJob) {
    textToSpeech.say("Hi there!")
}

firstPersonJob.cancel()
```
In this case, not only the "Hello!" launched with the firstPersonJob will get stopped, but the "Hi there!" launched with the secondPersonJob too.

Sadly, the Text-to-Speech API doesn't have a method for stopping a single synthesis from executing. I propose adding a parameter `clearQueueOnCancellation: Boolean` so that this behaviour remains optional. That would make this method  signature look like this:
```Kotlin
override suspend fun say(text: String, clearQueue: Boolean, clearQueueOnCancellation: Boolean = false)
```
and then with 
```Kotlin
cont.invokeOnCancellation {
    if (clearQueueOnCancellation) {
        stop()
    }
}
```

Reply:
Yes, agree with your suggestion. I have added the changes in following commit db1223940f117deae311ef234632b53800cdf093